dashboard/app/views/projects/index.html.haml

```diff
@@ -69,6 +69,10 @@
       $scope.projectsLoaded = true;
     }).catch($scope.genericError);
 
+    $scope.projectVisible = function(project) {
+      return (!project.hidden);
```

Comment:
if the service doesn't return projects that are 'hidden' then we can't use those projects on the client side (ie. we can't load the code from the project, we can't save them, etc).

Note that "client" doesn't mean "user"... the client here is code that we are writing.